loopd/swapclient_server.go

```diff
@@ -147,15 +147,20 @@ func (s *swapClientServer) Monitor(in *looprpc.MonitorRequest,
 
 	log.Infof("Monitor request received")
 
-	send := func(info loop.SwapInfo) error {
-		rpcSwap, err := s.marshallSwap(&info)
+	send := func(info *loop.SwapInfo) error {
+		rpcSwap, err := s.marshallSwap(info)
 		if err != nil {
 			return err
 		}
 
 		return server.Send(rpcSwap)
 	}
 
+	swaps, err := s.impl.FetchSwaps()
```

Comment:
Previously the in-memory swap snapshot was updated inside the same lock in which the notification was pushed out. This made sure that we didn't miss an update, or sent a duplicate. How is that situation now?

Reply:
Good point. I moved the fetch into the lock as well.